exec/java-exec/src/main/java/org/apache/drill/exec/record/RecordBatchSizer.java

```diff
@@ -525,4 +763,11 @@ public VectorInitializer buildVectorInitializer() {
     }
     return initializer;
   }
+
+  public void allocateVectors(VectorContainer container, int recordCount) {
+    for (VectorWrapper w : container) {
+      ColumnSize colSize = columnSizes.get(w.getField().getName());
+      colSize.allocateVector(w.getValueVector(), recordCount);
+    }
+  }
```

Comment:
Very nice clean-up and enhancements. Now I don't feel so bad about this stuff being quick & dirty...